Sources/Networking/Misc/DownloadState.swift

```diff
@@ -0,0 +1,14 @@
+//
+//  DownloadState.swift
+//  
+//
+//  Created by Matej Molnár on 16.01.2023.
+//
+
+import Foundation
+
+public enum DownloadState {
+    case progress(downloadedBytes: Double, totalBytes: Double)
+    case completed(Data?)
```

Comment:
i feel a bit confusion in states completed and terminated... when completed - that means finished with success we have Data as optional - so not sure if there are any... but for terminated - which I understand is fail we have Data type - so when we fail we always have data (to be able to continue) but when we finish we are not sure if we have data...